marshmallow_dataclass/union_field.py

```diff
@@ -39,6 +39,10 @@ def _serialize(self, value: Any, attr: str, obj, **kwargs) -> Any:
         if value is None:
             return value
         for typ, field in self.union_fields:
+            if typ.__dict__.get("__supertype__") and typ.__dict__.get(
+                "_marshmallow_field"
+            ):
+                typ = typ.__supertype__
```

Comment:
checks for `_marshmallow_field` to make sure this is only applied to marshmallow types